chromium_src/net/base/host_port_pair.cc

```diff
@@ -138,21 +130,22 @@ void HostPortPair::set_password(const std::string& in_password) {
   password_ = in_password;
 }
 
-bool HostPortPair::operator<(const HostPortPair& other) const {
-  if (HostPortPair_ChromiumImpl::operator<(other)) {
-    return true;
-  }
-  return std::tie(username_, password_) <
```

Comment:
Here is the main problem: if `*this > other ` we should return false, but we fallback to `(username_, password_)` comparison resulting in an inconsistent implementation.